bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -251,7 +255,7 @@ public Rule add(Rule rule) {
      * @param rule candidate for unique ID
      * @return a rule with UID
      */
-    protected Rule initRuleId(String rUID, Rule rule) {
+    protected @Nonnull Rule initRuleId(String rUID, Rule rule) {
```

Comment:
About the JavaDoc and annotation: The JavaDoc is for the developer. If a developer does not read the documentation he does it wrong. But: The nullness annotations are for the tooling on the development and compilation time. Tools (e.g. IDE or something that is used at compile time) could check if your written code is safe.